qiskit/transpiler/target.py

```diff
@@ -0,0 +1,600 @@
+            return None
+        return set(self._qarg_gate_map)
+
+    def get_qargs(self, gate):
+        """Get the qargs for a given gate name
+
+        Args:
+           gate (str): The gate instance to get qargs for
+        Returns:
+            set: The set of qargs the gate instance applies to
+        """
+        if None in self._gate_map:
+            return None
+        return set(self._gate_map[gate])
+
+    def durations(self):
+        """Get an InstructionDurations object from the target
+
+        Returns:
+            InstructionDurations: The instruction duration represented in the
+                target
+        """
+        if self._instruction_durations is not None:
+            return self._instruction_durations
+        out_durations = []
+        for instruction, props_map in self._gate_map.items():
+            for qarg, properties in props_map.items():
+                if properties is not None and properties.length is not None:
+                    out_durations.append((instruction, list(qarg), properties.length, "s"))
+        self._instruction_durations = InstructionDurations(out_durations)
+        return self._instruction_durations
+
+    def timing_constraints(self):
+        """Get an :class:`~qiskit.transpiler.TimingConstraints` object from the target
+
+        Returns:
+            TimingConstraints: The timing constraints represented in the Target
+        """
+        return TimingConstraints(
+            self.granularity, self.min_length, self.pulse_alignment, self.aquire_alignment
+        )
+
+    def instruction_schedule_map(self):
+        """Return an :class:`~qiskit.pulse.InstructionScheduleMap` for the
+        instructions in the target with a pulse schedule defined.
+
+        Returns:
+            InstructionScheduleMap: The instruction schedule map for the
+            instructions in this target with a pulse schedule defined.
+        """
+        if self._instruction_schedule_map is not None:
+            return self._instruction_schedule_map
+        out_inst_schedule_map = InstructionScheduleMap()
+        for instruction, qargs in self._gate_map.items():
+            for qarg, properties in qargs.items():
+                if properties is not None and properties.schedule is not None:
+                    out_inst_schedule_map.add(instruction, qarg, properties.schedule)
+        self._instruction_schedule_map = out_inst_schedule_map
+        return out_inst_schedule_map
+
+    def get_instruction_from_name(self, instruction):
+        """Get the :class:`~qiskit.circuit.Instruction` object for a given name
+
+        Args:
+            instruction (str): The instruction name to get the
+                :class:`~qiskit.circuit.Instruction` instance for
+        Returns:
+            Instruction: The Instruction instance corresponding to the name
+        """
+        return self._gate_name_map[instruction]
+
+    def get_instructions_for_qargs(self, qarg):
```

Comment:
can we call it `qubits` instead of `qargs`? :)

Reply:
Sure, I can rename it. I only called it qargs because it's the tuple `(0,)` not the qubits `0` and order is significant for multiqubit gates

Reply:
I am not sure why we have `qargs` and `qubits` in the codebase. I know in quantum_info Chris intentionally made it `qargs` because those classes are designed to work with not just 2-level qubits (e.g. it can be qutrits, ququarts, etc.).
But everywhere in the circuit and backend we only support qubits for now. So backend has `num_qubits` for example. But I think in some places in circuits we have a mix of qubits and qargs which makes it hard to remember